pkg/boot/mudp/mudp.go

```diff
@@ -0,0 +1,381 @@
+package mudp
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+	"net"
+	"sync"
+	"time"
+
+	capnp "capnproto.org/go/capnp/v3"
+	"github.com/cstockton/go-conv"
+	"github.com/libp2p/go-libp2p-core/discovery"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-core/peerstore"
+	"github.com/libp2p/go-libp2p-core/record"
+	cpMudp "github.com/wetware/casm/internal/api/mudp"
+)
+
+const (
+	multicastAddr = "224.0.1.241:3037"
+
+	discLimit = 10
+	discTTL   = time.Minute
+	discDist  = uint8(255)
+
+	timeout = 5 * time.Second
+)
+
+type Mudp struct {
+	h             host.Host
+	mc            *multicaster
+	disc          discovery.Discoverer
+	mustFind      map[string]chan peer.AddrInfo
+	mustAdvertise map[string]chan time.Duration
+	mu            sync.Mutex
+}
+
+func NewMudp(h host.Host, disc discovery.Discoverer) (mudp *Mudp, err error) {
+	mc, err := NewMulticaster(multicastAddr)
```

Comment:
I believe this will shadow the `err` variable, which would cause you to return `nil` in on line 43.